### PR TITLE
AEA-1401 implement dialogue type in aealite

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Implement the dialogue type in aealite.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

**Review by DavidMinarsch — APPROVED** [review]
LGTM so far

File: libs/go/aealite/protocols/dialogue_test.go [hunk 1]
```diff
@@ -1,178 +1,108 @@
...
-	counterPartyAddress = "1ba5cb6f46f426a27ec53064032419f1"
-	starterReference    = "starterReference"
-	responderReference  = "responderReference"
+	senderAddress       Address = "ba6b08b13043e83a962a3a5eeaad3b6c"
+	counterPartyAddress Address = "1ba5cb6f46f426a27ec53064032419f1"
+	starterReference    string  = "starterReference"
+	responderReference  string  = "responderReference"
 )
 
-// get a default dialogue label for testing purposes
```
Comment (marcofavorito) [inline libs/go/aealite/protocols/dialogue_test.go:16]: moved in `dialogue_label_test.go`

**Review by solarw — APPROVED** [review]
LGTM

File: libs/go/aealite/protocols/message.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+func (message *DialogueMessageWrapper) DialogueReference() DialogueReference {
+	return message.dialogueReference
+}
+
+func (message *DialogueMessageWrapper) Target() MessageId {
+	return message.target
+}
+
+func (message *DialogueMessageWrapper) Performative() Performative {
+	return message.body["performative"].(Performative)
```
Comment (marcofavorito) [inline libs/go/aealite/protocols/message.go:128]: this is a type assertion in Golang. It might fail, but if we `InitFromProtobuf` then the assert should pass (`body` is not exposed and `Body()` returns a copy of it)

File: libs/go/aealite/protocols/message.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	Target() MessageId
+	Performative() Performative
+	Body() map[string]interface{}
+	HasSender() bool
+	HasTo() bool
+	GetField(name string) interface{}
+	//ValidPerformatives() []Performative TODO temporarily removed
+}
+
+type DialogueMessageWrapper struct {
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply libs/go/aealite/protocols/message.go:48]: Because `DialogueMessage` has a Content field which is of type `byte[]`; that would mean to access an attribute it is required to parse the content of `byte[]` and access the required field. Also, that has to be done manually, as the user is interacting directly with `DialogueMessage`.

With this class, instead, we would do it only once, and access the fields saving the cost of decoding the `Content` field each time.

File: libs/go/aealite/protocols/dialogue.go [hunk 4]
```diff
@@ -563,15 +553,33 @@ func validateDialogueLabelExistence(dialogueLabel DialogueLabel) bool {
...
+		messageId:         messageId,
+		target:            target,
+		performative:      performative,
+		to:                counterParty,
+		sender:            selfAddress,
+		message:           content,
+	}
+	return initialMessage
 }
+*/
```
Comment (marcofavorito) [inline libs/go/aealite/protocols/dialogue.go:585]: this will be addressed in another PR (for `Dialogues`)

**Review by DavidMinarsch — APPROVED** [review]
LGTM
